common/stage.py

```diff
@@ -28,6 +28,7 @@
 import os
 import re
 import sys
+import tempfile
```

Comment:
This is required for platform independent finding of the temp folder. Ideally this would of course be overwritten by the command line parameter.

Reply:
Hmm, I think this is actually ok.
It might be faster to deal with internally in memory (if you haven't noticed the compiler keeps all states of all modifications in memory till the very end; this is so you can have fancy error messages).
But for now, this looks good.